libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.m

```diff
@@ -0,0 +1,301 @@
+    container.window.windowLevel = self.mainWindow.window.windowLevel + SFWindowLevelSnapshotOffset;
+    return container;
+}
+
+- (SFSDKWindowContainer *)passcodeWindow {
+    SFSDKWindowContainer *container = [self.namedWindows objectForKey:kSFPasscodeWindowKey];
+    if (!container) {
+        container = [self createPasscodeWindow];
+    }
+    //enforce WindowLevel
+    container.window.windowLevel = self.mainWindow.window.windowLevel + SFWindowLevelPasscodeOffset;
+    return container;
+}
+
+- (SFSDKWindowContainer *)createNewNamedWindow:(NSString *)windowName {
+    SFSDKWindowContainer * container = nil;
+    if ( ![self isReservedName:windowName] ) {
+        UIWindow *window = [self createDefaultUIWindow];
+        container = [[SFSDKWindowContainer alloc] initWithWindow:window name:windowName];
+        container.windowDelegate = self;
+        container.window.windowLevel = UIWindowLevelNormal;
+        container.windowType = SFSDKWindowTypeOther;
+        [self.namedWindows setObject:container forKey:windowName];
+    }
+    return container;
+}
+
+- (BOOL)isReservedName:(NSString *) windowName {
+    return ([windowName isEqualToString:kSFMainWindowKey] ||
+            [windowName isEqualToString:kSFLoginWindowKey] ||
+            [windowName isEqualToString:kSFPasscodeWindowKey] ||
+            [windowName isEqualToString:kSFSnaphotWindowKey]);
+    
+}
+
+- (BOOL)removeNamedWindow:(NSString *)windowName {
+    BOOL result = NO;
+    if (![self isReservedName:windowName]) {
+        [self.namedWindows removeObjectForKey:windowName];
+        result = YES;
+    }
+    return result;
+}
+
+- (SFSDKWindowContainer *)windowWithName:(NSString *)name {
+    return [self.namedWindows objectForKey:name];
+}
+
+- (void)addDelegate:(id<SFSDKWindowManagerDelegate>)delegate
+{
+    @synchronized (self) {
+        [_delegates addObject:[NSValue valueWithNonretainedObject:delegate]];
+    }
+}
+
+- (void)removeDelegate:(id<SFSDKWindowManagerDelegate>)delegate
+{
+    @synchronized (self) {
+        [_delegates removeObject:[NSValue valueWithNonretainedObject:delegate]];
+    }
+}
+
+- (void)enumerateDelegates:(void (^)(id<SFSDKWindowManagerDelegate> delegate))block
+{
+    @synchronized(self) {
+        [_delegates.allObjects enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+            id<SFSDKWindowManagerDelegate> delegate = [obj nonretainedObjectValue];
+            if (delegate) {
+                if (block) block(delegate);
+            }
+        }];
+    }
+}
+#pragma mark - SFSDKWindowContainerDelegate
+- (void)windowEnable:(SFSDKWindowContainer *_Nonnull)window animated:(BOOL)animated withCompletion:(void (^)(void))completion{
+
+    [self enumerateDelegates:^(id<SFSDKWindowManagerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowManager:willEnableWindow:)]){
+            [delegate windowManager:self willEnableWindow:window];
+        }
+    }];
+    
+    if (animated) {
```

Comment:
Instead of just making the window we want to show opaque, could we also make all the other ones transparent. That way the window we are trying to enable, would be the one that ends up being shown/active.